timemachine/cpp/src/gpu_utils.cuh

```diff
@@ -34,10 +34,22 @@ inline void curandAssert(curandStatus_t code, const char *file, int line, bool a
     }
 }
 
+/* cudaSafeMalloc is equivalent to gpuErrchk(cudaMalloc(...)) except that it prints a warning message if the
+* allocation is greater than a GiB.
+*/
+#define cudaSafeMalloc(ptr, size)                                                                                      \
+    ({                                                                                                                 \
+        if (size > (1 << 30)) {                                                                                        \
+            fprintf(stderr, "cudaSafeMalloc: allocation larger than 1GiB %s %d\n", __FILE__, __LINE__);                \
+        }                                                                                                              \
+        cudaError_t code = cudaMalloc(ptr, size);                                                                      \
```

Comment:
nit: might be preferable to obfuscate `code` as something like `cudaSafeMalloc__code` (or just inline this definition to the next line) to prevent variable capture in the unlikely case that one of the macro args is `code`

Reply:
Hmmm makes me think templating is the right way to do it, or at the very least move the cudaMalloc call into the gpuAssert to avoid the collision. Will play around with it.

Reply:
Fixed in [ac06ffe](https://github.com/proteneer/timemachine/pull/899/commits/ac06ffe94bd1fa6d1d75fcb235dd6e03d6f8f419)